pallets/dca/src/lib.rs

```diff
@@ -0,0 +1,1023 @@
+			Self::ensure_that_sell_amount_is_bigger_than_transaction_fee(&schedule)?;
+
+			let next_schedule_id = Self::get_next_schedule_id()?;
+
+			Schedules::<T>::insert(next_schedule_id, &schedule);
+			ScheduleOwnership::<T>::insert(who.clone(),next_schedule_id,());
+			RemainingAmounts::<T>::insert(next_schedule_id,schedule.total_amount);
+
+			Self::reserve_named_reserve(&schedule, &who)?;
+
+			let blocknumber_for_first_schedule_execution =
+				start_execution_block.unwrap_or_else(|| Self::get_next_block_number());
+			Self::plan_schedule_for_block(blocknumber_for_first_schedule_execution, next_schedule_id)?;
+
+			Self::deposit_event(Event::Scheduled {
+				id: next_schedule_id,
+				who: who.clone(),
+			});
+			Self::deposit_event(Event::ExecutionPlanned {
+				id: next_schedule_id,
+				who,
+				block: blocknumber_for_first_schedule_execution,
+			});
+
+			Ok(())
+		}
+
+		/// Pause the DCA schedule planned in the given block number
+		///
+		/// Parameters:
+		/// - `origin`: schedule owner
+		/// - `schedule_id`: schedule id
+		/// - `next_execution_block`: block number where the DCA is planned to be executed
+		///
+		/// Emits `Paused` event when successful.
+		///
+		#[pallet::weight(<T as Config>::WeightInfo::pause())]
+		#[transactional]
+		pub fn pause(
+			origin: OriginFor<T>,
+			schedule_id: ScheduleId,
+			next_execution_block: BlockNumberFor<T>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+			Self::ensure_that_origin_is_schedule_owner(schedule_id, &who)?;
+
+			Self::remove_schedule_id_from_next_execution_block(schedule_id, next_execution_block)?;
+			Suspended::<T>::insert(schedule_id, ());
+
+			Self::deposit_event(Event::Paused { id: schedule_id, who });
+
+			Ok(())
+		}
+
+		/// Resume the suspended DCA schedule for the specified next execution block number
+		/// If next execution block number is not specified, then the schedule is planned in the consequent block
+		///
+		/// Parameters:
+		/// - `origin`: schedule owner
+		/// - `schedule_id`: schedule id
+		/// - `next_execution_block`: block number to plan the next execution of the schedule.
+		///
+		/// Emits `Resumed`and `ExecutionPlanned` event when successful.
+		///
+		#[pallet::weight(<T as Config>::WeightInfo::resume())]
+		#[transactional]
+		pub fn resume(
+			origin: OriginFor<T>,
+			schedule_id: ScheduleId,
+			next_execution_block: Option<BlockNumberFor<T>>,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+			Self::ensure_that_origin_is_schedule_owner(schedule_id, &who)?;
+			Self::ensure_that_next_blocknumber_is_bigger_than_current_block(next_execution_block)?;
+			Self::ensure_that_schedule_is_suspended(schedule_id)?;
+
+			let next_execution_block = next_execution_block.unwrap_or_else(|| Self::get_next_block_number());
+			Self::plan_schedule_for_block(next_execution_block, schedule_id)?;
+
+			Suspended::<T>::remove(schedule_id);
```

Comment:
@mrq1911 suggested that the schedules storage should not changed at all, therefore we track it separately. Because otherwise we would need to write in that storage and since it can be a big storage, it might be less performant

Reply:
My assumption would be that you would want to optimize the code that is executed the most. Which IME is schedule execution.

Reply:
as suspended feature will be removed, this comment is not applicable anymore.